### ISSUE TITLE
Model predicts 100% Wake on 9-hour recording (non-US equipment) - Need guidance

### ISSUE THREAD

**Issue by dfguerrerom** [zou-group/sleepfm-clinical#11]
## The Problem

SleepFM predicts 100% Wake on a full-night PSG despite clinical report showing 97.8% sleep efficiency.

| What we got | Clinical ground truth |
|-------------|----------------------|
| 100% Wake, 0% Sleep | 24% REM, 25% N3, 44% N2, 5% N1 |

### Key Finding

The model IS processing signals - Wake probability varies from 58% to 99% throughout the night. But it never drops below 50%, so argmax always picks Wake.

```
Time        Wake    REM     N1      N2      N3
21:19       97.9%   0.5%    0.9%    0.4%    0.3%   (patient awake - correct!)
02:19       79.5%   6.3%    12.4%   0.6%    1.3%   (should be deep sleep)
```

This suggests a domain shift issue, not a preprocessing error.

### Our Setup

- **Equipment**: Neurovirtual BWAnalysis
- **Preprocessing**: Used your `EDFToHDF5Converter` exactly as documented
- **Verified**: 128 Hz, z-score normalized, correct HDF5 structure
- **Channels**: 41 mapped to BAS/EKG/RESP/EMG per `channel_groups.json`

### Questions

1. Is SleepFM expected to generalize to non-US equipment without fine-tuning?
2. Any preprocessing steps we might be missing for international equipment?
3. Is there a recommended way to calibrate probabilities for new data domains?

We're happy to share our data for debugging or contribute to documentation for non-US compatibility.

---

### Full technical details

**Environment:**
- Recording: 8.9 hours, 57 channels, converted from proprietary .psg
- GPU: NVIDIA A10G, PyTorch 2.9.0+cu128

**Channel mapping:**
- BAS: 20 channels (EEG F3-M2, C3-M2, EOG LOC/ROC, etc.)
- EKG: 5 channels (ECG I, II, III, AVR)
- RESP: 10 channels (Flow, SpO2, Effort Thor/Abdo)
- EMG: 6 channels (Chin, LAT, RAT, Leg)

**What we verified:**
- [x] Sample rate: 128 Hz
- [x] Z-score normalization applied
- [x] HDF5 structure matches expected format
- [x] All 4 modalities have valid data
- [x] Tested on CPU and GPU (same results)

**Disease model (`model_diagnosis`):** Also shows calibration issues - 675/1065 diseases with >90% risk.

**Comment by rthapa84** [comment]
Hello,

Thank you for reaching out and for the detailed experiments.

From what you describe, this does appear more consistent with a calibration / domain-shift issue than a preprocessing failure. We have not yet systematically validated calibration on international acquisition systems without adaptation, so some degradation on new hardware is unfortunately possible.

As a quick diagnostic, you could ignore argmax and instead plot the full probability trajectories. If the hypnogram becomes qualitatively reasonable after smoothing or thresholding, that would strongly suggest miscalibration rather than complete model failure.

If you have access to a small labeled subset, there are two promising directions:

(1) Post-hoc calibration, such as temperature scaling or class-wise bias correction, applied on a held-out validation set.

(2) Lightweight adaptation / fine-tuning, for example freezing the backbone and re-training only the final classification heads on local data with a low learning rate and short training schedule. You may need to play around with hyperparameters a bit.

Even a modest amount of annotated data can materially improve stage balance and disease-risk calibration.

**Comment by unidesigner** [comment]
@dfguerrerom Have you managed to calibrate your data to get meaningful results?

**Comment by 1-wang-lei** [comment]
怎么解决的100%wake

**Comment by klat0n** [comment]
Hello,  I too am having the same results.  Has there been any updates on a solution or new model generated?
